### ISSUE TITLE
We should only keep a version of `init_fourier_space` returning a `FourierSampling` in our public API.

### ISSUE THREAD

**Issue by jbigot** [CExA-project/ddc#333]
@blegouix can maybe answer better than me on this but if I remember correctly the idea was to ease initialization of multiple dimensions at once. I think the only place where it is used in a variadic context is the DDC tests. It could be moved to the tests so that we only keep a version returning a `FourierSampling` in our public API.

_Originally posted by @tpadioleau in https://github.com/CExA-project/ddc/pull/121#discussion_r1517362751_

**Comment by jbigot** [comment]
> In a future PR we should uniformize how initializations are done

_Originally posted by @jbigot in https://github.com/CExA-project/ddc/pull/121#discussion_r1510677273_

**Comment by blegouix** [comment]
I agree with the Thomas' comment I think I wanted to permit the simple `ddc::init_fourier_space<X...>(x_mesh);`, but now that I am more used to DDC I see it does not corresponds to our usual conventions to initialize dimensions.
